### ISSUE TITLE
VSCode 1.78, 1.79: Fatal error: exception Failure("Couldn't connect on 127.0.0.1:6000")

### ISSUE THREAD

**Issue by djaonline** [vshaxe/vshaxe#589]
os: linux 
haxe: 4.0.5
vscode: 1.78.2
Compilation server doesn't connect in vscode 1.78.2. After compiling haxe hello world project:  
```
Executing task: haxe --connect 6000 build.hxml 
Fatal error: exception Failure("Couldn't connect on 127.0.0.1:6000")
```
Haxe output: 
```
Haxe language server started
Haxe Path: haxe
Using --server-connect
Haxe connected!
Listening on port 6000
Done.
```

vscode 1.77.3 works ok.

**Comment by djaonline** [comment]
@RblSb Yes, please https://github.com/djaonline/haxe-hello-world

**Comment by RblSb** [comment]
Cannot repro on 4.0.5/4.2.5 and different vahaxe versions on latest vscode. This can be platform specific vscode problem, or something else is different in environment. We probably should make some verbose mode for debugging this.
I only can get this error in Kha, when i use watch mode. But it only happens on first compilation, and then recompiles normally.
Comes from this exception:
https://github.com/HaxeFoundation/haxe/blob/baee98f4be27d8c4b63620535e1d20284d831cab/src/compiler/server.ml#L542
Maybe there is different error types tho, so we can show more specific error messages.

**Comment by djaonline** [comment]
Vscode 1.78 updated to Electron 22. Possible tne problem is here.

**Comment by AlexHaxe** [comment]
well, key do fixing things is a reproducible sample. you've provided a sample, however it fails to reproduce on RblSb and my systems. since it's also a very basic sample that "should" just work, an issue like yours should affect everyone and there would be more reports of it. so chances are it is a problem with your local configuration / setup. 

- try disabling extensions and see if it starts working. 
- install an insider build of VSCode, they use separate configuration / extensions folders; install only vshaxe extension and try to run your project
- make sure there is no antivirus or firewall at play, interfering with Haxe / VSCode operations
-try reproducing with a different user login or on a separate machine (if available), or try booting a from Linux Live DVD and see if it works

**Comment by djaonline** [comment]
By the way if I run haxe server manually `haxe -v --wait 6000` it starts work correctly.

**Comment by kLabz** [comment]
Sounds like your vscode is using a wrong haxe executable, possibly because of a misconfiguration you haven't shared.

**Comment by djaonline** [comment]
@kLabz  I use for that porpose your project https://github.com/kLabz/haxe-manager (very nice tool, thank you :)  )
Here its path: `/home/djaonline/tools/haxe-manager/`
Env variables:
```
export MY_TOOLS=$HOME/tools
export HAXE_STD_PATH=$MY_TOOLS/haxe-manager/std
export HAXELIB_PATH=$MY_TOOLS/haxe-manager/haxelibs
export NEKOPATH=/usr/lib/neko
export LD_LIBRARY_PATH=/usr/lib/neko
```

**Comment by AlexHaxe** [comment]
can you confirm that localhost points to `::1` in your hosts file? and if you let it point to 127.0.0.1 will it fix your issue?

**Comment by djaonline** [comment]
@AlexHaxe  The current host file record is `::1 localhost.localdomain   localhost ip6-localhost`. Changing it to `::1                     localhost.localdomain   127.0.0.1` solves the issue.

**Comment by AlexHaxe** [comment]
closed by #605

**Comment by AlexHaxe** [comment]
vshaxe version 2.30.0 was just released. it should solve your issue.
